subprojects/base-services/src/test/groovy/org/gradle/api/JavaVersionSpec.groovy

```diff
@@ -57,10 +58,13 @@ public class JavaVersionSpec extends Specification {
         JavaVersion.toVersion("7") == JavaVersion.VERSION_1_7
         JavaVersion.toVersion("8") == JavaVersion.VERSION_1_8
         JavaVersion.toVersion("9") == JavaVersion.VERSION_1_9
+        JavaVersion.toVersion("10") == JavaVersion.VERSION_1_10
 
         JavaVersion.toVersion("1.9.0-internal") == JavaVersion.VERSION_1_9
         JavaVersion.toVersion("1.9.0-ea") == JavaVersion.VERSION_1_9
         JavaVersion.toVersion("9-ea") == JavaVersion.VERSION_1_9
+
+        JavaVersion.toVersion("10-ea") == JavaVersion.VERSION_1_10
```

Comment:
Nitpick that we might test with "10-internal" because that's currently the version from the example.